LogMonitor/src/LogMonitor/EtwMonitor.cpp

```diff
@@ -1287,6 +1288,39 @@ EtwMonitor::GetPropertyLength(
             {
                 PropertyLength = EventInfo->EventPropertyInfoArray[Index].length;
             }
+            else if (0 == (EventInfo->EventPropertyInfoArray[Index].Flags & (PropertyStruct | PropertyParamCount)) &&
+                EventInfo->EventPropertyInfoArray[Index].count == 1 )
+            {
+                BYTE const* pbData = static_cast<BYTE const*>(EventRecord->UserData);
+                BYTE const* pbDataEnd = pbData + EventRecord->UserDataLength;
+
+                switch (EventInfo->EventPropertyInfoArray[Index].nonStructType.InType)
+                {

```

Comment:
This rest bit of the addition is inspired with how this function is implemented: https://learn.microsoft.com/en-us/windows/win32/etw/using-tdhformatproperty-to-consume-event-data#:~:text=void%20PrintProperties(unsigned%20propBegin%2C%20unsigned%20propEnd)